src/components/button.js

```diff
@@ -6,53 +6,106 @@
-            : undefined
-        }
-        classes={{
-          root: classes.root,
-          contained: classes.contained,
-          outlined: classes.outlined,
-        }}
-        className={[
-          visible || isDev ? '' : classes.hide,
-          options.buttonText.length === 0 ? classes.empty : '',
-        ].join(' ')}
-        // only set submit when submit prefab is used
-        type={isDev ? 'button' : type}
-        //        type={linkType === undefined ? undefined : 'submit'}
-        // if prefab is button set href when link is external
-        href={linkType === 'External' ? linkToExternal : undefined}
-        // if prefab is button and internal link
-        component={linkType === 'Internal' ? B.Link : undefined}
-        endpoint={linkType === 'Internal' ? linkTo : undefined}
-      >
-        {options.buttonText}
-      </Button>
+    const isAction = linkType === 'action';
+
+    const generalProps = {
+      disabled,
+      size,
+      href: linkType === 'external' ? linkToExternal : undefined,
+      component: linkType === 'internal' ? B.Link : undefined,
+      endpoint: linkType === 'internal' ? linkTo : undefined,
+    };
+
+    const iconButtonProps = {
+      ...generalProps,
+      classes: { root: classes.root },
+      classname: visible || isDev ? '' : classes.hide,
+    };
+
+    const buttonProps = {
+      ...generalProps,
+      fullWidth,
+      variant,
+      [`${iconPosition}Icon`]:
+        icon !== 'None' ? React.createElement(Icons[icon]) : undefined,
+      classes: {
+        root: classes.root,
+        contained: classes.contained,
+        outlined: classes.outlined,
+      },
+      className: [
+        visible || isDev ? '' : classes.hide,
+        buttonText.length === 0 ? classes.empty : '',
+      ].join(' '),
+      type: isDev ? 'button' : type,
+    };
+
+    let ButtonComponent =
+      variant === 'icon' ? (
+        <IconButton {...iconButtonProps}>
+          {React.createElement(Icons[icon === 'None' ? 'Error' : icon], {
+            fontSize: size,
+          })}
+        </IconButton>
+      ) : (
+        <Button {...buttonProps}>{buttonText}</Button>
+      );
+
+    const Loader = (
+      <CircularProgress size={16} className={classes.loader}></CircularProgress>
     );
+
+    if (isAction) {
+      ButtonComponent = (
+        <B.Action actionId={actionId}>
+          {(callAction, { loading }) => {
+            const onClickAction = event => {
+              event.preventDefault();
+              if (!isDev && !loading && linkType === 'action') callAction();
+            };
+            const actionClickHandler = isAction && { onClick: onClickAction };
+            return variant === 'icon' ? (
+              <IconButton {...iconButtonProps} {...actionClickHandler}>
+                {loading
+                  ? Loader
+                  : React.createElement(
+                      Icons[icon === 'None' ? 'Error' : icon],
```

Comment:
The default (when not loading) is error. This means that the button is never going to get a success state? It always is meant to redirect? Why is this only in the icon button?

Reply:
That is the default Icon when there is no icon selected on the Icon dropdown option.

Seeing this component supports both Button and Icon we need to support the option "None"

Reply:
Regarding the success state, you're right that hasn't been added